clang/test/Interpreter/private-member-access.cpp

```diff
@@ -0,0 +1,35 @@
+// RUN: cat %s | clang-repl | FileCheck %s
+
+extern "C" int printf(const char*, ...);
+
+// Test 1: Pointer to private type alias (the original bug report)
+class io_context { using impl_type = int; public: impl_type* get(); };
+io_context::impl_type* io_context::get() { return nullptr; }
+printf("Pointer to private type: %s\n", io_context().get() == nullptr ? "passed" : "failed");
```

Comment:
Do we need these printf? Won't we obtain a diagnostic because the access will fail? Can we just check that?